Source/KSCrash/Source/KSCrash/Recording/Tools/BSG_KSFileUtils.c

```diff
@@ -75,19 +75,14 @@ bool bsg_ksfuflushWriteBuffer(const int fd) {
 
 bool bsg_ksfuwriteBytesToFD(const int fd, const char *const bytes,
                             ssize_t length) {
-    ssize_t newLen = bufferLen + length;
-
-    if (newLen >= BUFFER_SIZE) { // flush the current buffer
-        bsg_ksfuflushWriteBuffer(fd);
-        newLen = bufferLen + length; // recalculate new position
-    }
-
-    for (ssize_t k = bufferLen; k < newLen;
-         k++) { // place in buffer for future write
-        ssize_t j = k - bufferLen;
-        charBuffer[k] = bytes[j];
+    
+    for (ssize_t k = 0; k < length; k++) {
+        if (bufferLen >= BUFFER_SIZE) {
+            bsg_ksfuflushWriteBuffer(fd); // flush to disk
```

Comment:
If `bsg_ksfuflushWriteBuffer` fails and returns `false`, `bufferLen` won't be updated, and then it'll overflow on the next write to `charBuffer `. Maybe we should bail out and return `false` in that case?

Reply:
Ah @benasher44 beat me to it.  Yeah I realized this function always returns `true`